tests/test_dirs.py

```diff
@@ -0,0 +1,276 @@
+"""
+Copyright (c) 2021 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+import os
+import tempfile
+from pathlib import Path
+from typing import Any, Dict, Iterable
+
+import pytest
+
+from atomic_reactor.constants import DOCKERFILE_FILENAME, Platform
+from atomic_reactor.dirs import DockerfileNotExist, BuildDir, RootBuildDir, ImageInspectionData
+from atomic_reactor.source import DummySource
+from dockerfile_parse import DockerfileParser
+
+
+def test_builddir_failure_on_nonexisting_path():
+    with pytest.raises(FileNotFoundError, match="does not exist"):
+        BuildDir(Path("some_dir"), Platform.x86_64)
+
+
+def test_builddir_dockerfile_path(tmpdir):
+    dir_path = Path(tmpdir)
+    dir_path.joinpath(DOCKERFILE_FILENAME).touch()
+    build_dir = BuildDir(dir_path, Platform.x86_64)
+    assert Path(tmpdir.join(DOCKERFILE_FILENAME)) == build_dir.dockerfile_path
+
+
+def test_builddir_dockerfile_path_returns_absolute_path(tmpdir):
+    dir_path = Path(tmpdir)
+    dir_path.joinpath(DOCKERFILE_FILENAME).touch()
+    build_dir = BuildDir(dir_path, Platform.x86_64)
+    assert build_dir.dockerfile_path.is_absolute()
+
+
+def test_builddir_dockerfile_not_support_linked_dockerfile(tmpdir):
+    some_dir = Path(tmpdir)
+    dockerfile = some_dir / DOCKERFILE_FILENAME
+    dockerfile.touch()
+    # link the Dockerfile
+    build_dir = Path(tmpdir, f"build_{Platform.x86_64}")
+    build_dir.mkdir()
+    build_dir.joinpath(DOCKERFILE_FILENAME).symlink_to(dockerfile)
+    with pytest.raises(DockerfileNotExist, match="Dockerfile is linked from"):
+        print(BuildDir(build_dir, Platform.x86_64).dockerfile_path)
+
+
+def test_builddir_get_parsed_dockerfile(tmpdir):
+    dir_path = Path(tmpdir)
+    dir_path.joinpath(DOCKERFILE_FILENAME).write_text("FROM fedora:35", "utf-8")
+    build_dir = BuildDir(dir_path, Platform.x86_64)
+    assert isinstance(build_dir.dockerfile, DockerfileParser)
+
+
+@pytest.mark.parametrize("inspection_data,expected_envs", [
+    [{}, {"HOME": ""}],
+    [{"Env": {}}, {"HOME": ""}],
+    [{"Env": []}, {"HOME": ""}],
+    [
+        {"Env": ["HOME=/home", "var2=--option=first"]},
+        {"HOME": "/home"},
+    ],
+    [
+        {"Env": {"HOME": "/home", "var2": "--option=first"}},
+        {"HOME": "/home"},
+    ],
+])
+def test_builddir_get_parsed_dockerfile_with_parent_env(
+    inspection_data: Dict[str, Any], expected_envs, tmpdir
+):
+    dir_path = Path(tmpdir)
+    dir_path.joinpath(DOCKERFILE_FILENAME).write_text(
+        "FROM base-image\nENV HOME=$HOME", "utf-8"
+    )
+    build_dir = BuildDir(dir_path, Platform.x86_64)
+    inspection_data = ImageInspectionData(inspection_data)
```

Comment:
Oh, I didn't realize that `NewType` requires you to wrap the value in the new type. Would a type alias be enough? https://docs.python.org/3/library/typing.html#type-aliases

Reply:
Cool. Type alias is better in this case. :+1: